File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        Queue<TreeNode> que = new LinkedList<>();
        List<Integer> list = new ArrayList<>();
        if(root == null) return list;
        que.add(root);
        while(!que.isEmpty()){
            int size = que.size();
            for(int i = 0 ; i < size ; i++){
                if(que.peek().right != null){
                    que.add(que.peek().right);
                }
                if(que.peek().left != null){
                    que.add(que.peek().left);
                }
                if(i == 0)
                    list.add(que.poll().val);
                else que.poll();
            }
        }
        return list;
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        queue<TreeNode*> que ;
        if(root == nullptr){
            return {};
        }
        vector<int> v; 
        que.push(root);
        while(!que.empty()){
            int s = que.size();
            for(int i = 0 ; i < s ; i++){
                TreeNode* r = que.front();
                que.pop();
                if(r->right != nullptr){
                    que.push(r->right);
                }
                if(r->left != nullptr){
                    que.push(r->left);
                }
                if(i == 0){
                    v.push_back(r->val);
                }
            }
        }
        return v;
    }
};